libs/libc/semaphore/sem_init.c

```diff
@@ -74,7 +74,7 @@ int nxsem_init(FAR sem_t *sem, int pshared, unsigned int value)
       /* Initialize to support priority inheritance */
 
 #ifdef CONFIG_PRIORITY_INHERITANCE
-      sem->flags            = 0;
+      sem->flags           |= PRIOINHERIT_FLAGS_DISABLE;
```

Comment:
change to:
```
sem->flags           = PRIOINHERIT_FLAGS_DISABLE;
```
or change `PRIOINHERIT_FLAGS_DISABLE` to `PRIOINHERIT_FLAGS_ENABLE`